src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/NotifyReferralServiceTest.kt

```diff
@@ -100,4 +110,52 @@ class NotifyReferralServiceTest {
     assertThat(personalisationCaptor.firstValue["referenceNumber"]).isEqualTo("AJ9871AC")
     assertThat(personalisationCaptor.firstValue["referralUrl"]).isEqualTo("http://example.com/referral/42c7d267-0776-4272-a8e8-a673bfe30d0d")
   }
+
+  @Nested
+  inner class ReferralDetailsChangedEvent {
+    private val referralDetailsFactory = ReferralDetailsFactory()
+    private val referral = referralFactory.createAssigned()
+
+    private val referralDetailsChangedEvent = ReferralEvent(
+      "source",
+      ReferralEventType.DETAILS_CHANGED,
+      referral,
+      "http://localhost:8080/sent-referral/${referral.id}",
+      data = mapOf(
+        "newDetails" to ReferralDetailsDTO.from(
+          referralDetailsFactory.create(
+            referral.id,
+            referral.createdAt,
+            referral.createdBy,
+            completionDeadline = LocalDate.of(2022, 6, 6)
+          )
+        ),
+        "previousDetails" to ReferralDetailsDTO.from(
+          referralDetailsFactory.create(
+            referral.id,
+            referral.createdAt,
+            referral.createdBy,
+            completionDeadline = LocalDate.of(2022, 4, 13)
+          )
+        ),
+        "currentAssignee" to AuthUserDTO.from(referral.currentAssignee!!)
+      )
+    )
+
+    @Test
+    fun `referral details changed event sends email when completion deadline has changed`() {
```

Comment:
Can we add a test that 'sensitive' fields are not included in the email by accident in the future, please?

Reply:
Shall we also add a test about the assertion in the PR description?

> if the referral is unassigned, no email is sent.

Reply:
yep - good idea.

Reply:
found a bug relating to unassigned cases, so thanks for noting this.